models/dvmdostem/R/write.config.dvmdostem.R

```diff
@@ -0,0 +1,353 @@
+                  file.path(rundir, 'config'))))
+
+  if (dir.exists(file.path(rundir, 'parameters'))) {
+    unlink(file.path(rundir, 'parameters'), recursive=TRUE)
+  }
+  system2(paste0("cp"),
+          wait=TRUE,
+          args=(c("-r",
+                  file.path(appbinary_path, 'parameters'),
+                  file.path(rundir, 'parameters'))))
+
+  # (1)
+  # Read in a parameter data block from dvmdostem
+
+  # Build a dataframe from the incoming trait.values. trait.values
+  # contains the meta-analysis posteriors values for each parameter (trait)
+  # that we are hoping to "inject" into dvmdostem. Convert to a dataframe
+  # for easier indexing...
+  trait_df <- convert.samples.dvmdostem(trait.values)
+  ## !! TODO: Should run this as below so that only trait names remain !!
+  ## !! But if we do that now, the parsing below wont work.  Need to
+  ## !! separate parsing of PFT name for json and trait conversion for BETYdb
+  #trait_df <- convert.samples.dvmdostem(trait.samples = trait.values[[settings$pfts$pft$name]])
+  
+  # Now we have to read the approporate values out of the trait_df
+  # and get those values written into the parameter file(s) that dvmdostem will
+  # need when running. Because the dvmdostem parameters have a sort of
+  # interesting, semi-standardized, space delimited, column format, we'll
+  # use some helper scripts from dvmdostem that allow us to more easily
+  # handle the parameter files and write our new trait values into the correct
+  # place. The basic flow will be like this:
+  #  - Read dvmdostem parameter file into json object, load into memory
+  #  - Update the in-memory json object
+  #  - Write the json object back out to a new dvmdostem parameter file
+
+  # Start by figuring out which community number we are working on.
+  # This is important because we need to pull the correct "community block"
+  # from the dvmdostem parameter files. Each trait in the traits dataframe
+  # should have a name like this:
+  #   CMT04.Salix.extinction_coefficient_diffuse
+  # So we can parse the first name in the dataframe to determine the CMT number.
+  # !! Actually, it should just be CMT04.Salix OR extinction_coefficient_diffuse
+  # !! Can't have PFT name linked with trait otherwise that would require a lot of 
+  # !! hard coding in this function. 
+  cmtnum <- strsplit(unlist(strsplit(names(trait_df)[1], '.', fixed=TRUE))[1], "CMT")
+  cmtnum <- unlist(cmtnum)[2]
+  cmtnum <- as.numeric(cmtnum)
+
+  # Next, use a helper script distributed with dvmdostem to read the dvmdostem
+  # parameter data into memory as a json object, using a temporaroy json file
+  # to hold a representation of each dvmdostem parameter file.
+  dimveg_params <- paste(appbinary_path, "parameters", 'cmt_dimvegetation.txt', sep="/")
+  envcanopy_params <- paste(appbinary_path, "parameters", 'cmt_envcanopy.txt', sep="/")
+
+  dimveg_jsonfile <- '/tmp/dvmdostem-dimveg.json'
+  envcanopy_jsonfile <- '/tmp/dvmdostem-envcanopy.json'
```

Comment:
This seems dangerous, as different sites and ensemble members would all be using the same tmp file.